swap_meet/vendor.py

```diff
@@ -1,2 +1,79 @@
+from swap_meet.item import Item
+
 class Vendor:
-    pass
+    
+    def __init__(self, inventory=[], items=[]):
+        self.inventory = inventory
+        self.items = items 
+        
+    def add(self, item_to_add):
+        self.inventory.append(item_to_add)
+        return item_to_add
+    
+    def remove(self, item_to_remove):
+        if item_to_remove in  self.inventory:
+            self.inventory.remove(item_to_remove)
+            return item_to_remove
+        else:
+            return False
+        
+    def get_by_category(self, category):
+        item_list = []
+        for item in self.inventory:
+            if item.category == category:
+                item_list.append(item)
+        return item_list
+    
+    def swap_items(self, other_vendor, my_item, their_item):
+        if my_item in self.inventory and their_item in other_vendor.inventory:
```

Comment:
Consider reversing the sense of any logic like this where the else is a simple return so that the return is part of the if. By checking that condition and returning as soon as we detect that some minimum requirements are not met, we convert the if conditional into a "guard clause". Since it returns when run, we don't need to use the else keyword, and we can avoid indenting our main function logic.

This is a very common pattern in production code.